plugins/env/app/models/environment_variable.rb

```diff
@@ -64,14 +66,28 @@ def serialize(variables, env_scopes)
       end.join("\n")
     end
 
-    def config_service_folder(project, required:)
-      url = ENV["CONFIG_SERVICE_URL"]
-      raise KeyError, "CONFIG_SERVICE_URL not set" if required && !url
-      "#{url}/samson/#{project.permalink}"
+    def config_service_folder(project)
+      "samson/#{project.permalink}"
+    end
+
+    def config_service_label(project)
+      path = config_service_folder(project)
+      path = "s3://#{ENV["CONFIG_SERVICE_BUCKET"]}/#{path}" if ENV.key? "CONFIG_SERVICE_BUCKET"
+      path
+    end
+
+    def config_service_bucket(stub_responses: false)
+      bucket_name = ENV.fetch("CONFIG_SERVICE_BUCKET")
+      region = ENV.fetch("CONFIG_SERVICE_REGION")
+      Aws::S3::Resource.new(region: region, stub_responses: stub_responses).bucket(bucket_name)
```

Comment:
prefer
 - caching the s3 client
 - using the direct methods instead of the fancy object wrappers since they show which permissions are needed too

https://docs.aws.amazon.com/sdk-for-ruby/v3/api/Aws/S3/Client.html#get_object-instance_method

```
def config_service_read(key)
  @s3_client ||= Aws::S3::Client.new(region: region)
  s3_client.get_object(bucket: bucket_name, stub_responses: stub_responses).get_object(key).body.read
ebd
```